### ISSUE TITLE
hash binary files

### ISSUE THREAD

**Issue by slxcc** [digitalbazaar/forge#128]
the message digest class can't process raw binary read via FileReader.
An UTF-8 conversion is forced on append via this function (util.js:620)
util.ByteStringBuffer.prototype.toString = function() {
  return util.decodeUtf8(this.bytes());
};
we noticed a lot of movement in this area in the last release (and a bunch of TODOs in the source), is there work in progress on this part ?
(Our test used a jpeg image read via FileReader.readAsBinaryString(...))

If the Buffer part are in heavy refactoring we'd like to know because I also plan to subclass the output buffer to use with cipher to consume the output, and not keep all the results in memory at the end of an encrypt operation.

**Comment by ingnam2016** [comment]
How do we decrypt an encrypted message as encrypted message is not readable.

**Comment by davidlehn** [comment]
@ingnam2016: Can you rephrase the question?  This issue has been closed for ~4 years, if this is related to another topic, please open a new issue.  Thanks.

**Comment by ingnam2016** [comment]
Thanks @davidlehn, for the help, my issue has been resolved.
